torchquad/utils/set_precision.py

```diff
@@ -9,45 +9,77 @@ def set_precision(data_type="float"):
     """This function allows the user to set the default precision. Call before declaring your variables.
 
     Args:
-        data_type (string, optional): Data type to use, either "float" or "double". Defaults to "float".
+        data_type (string, optional): Data type to use, either "float"/"float32", "double"/"float64", "cfloat"/"complex64", or "cdouble"/"complex128". Defaults to "float".
 
     """
 
     if torch.cuda.is_initialized():
-        if data_type.lower() == "double":
+        if data_type.lower() == "double" or data_type.lower() == "float64":
             logging.info(
-                "Setting default tensor type to cuda.Float64 (CUDA is initialized)."
+                "Setting default tensor type to cuda.DoubleTensor (CUDA is initialized)."
             )
             torch.set_default_tensor_type(torch.cuda.DoubleTensor)
-        elif data_type.lower() == "float":
+        elif data_type.lower() == "float" or data_type.lower() == "float32":
             logging.info(
-                "Setting default tensor type to cuda.Float32 (CUDA is initialized)."
+                "Setting default tensor type to cuda.FloatTensor (CUDA is initialized)."
+            )
+            torch.set_default_tensor_type(torch.cuda.FloatTensor)
+        elif data_type.lower() == "cdouble" or data_type.lower() == "complex128":
+            # logging.info(
+            logging.error(
+                # "Setting default tensor type to cuda.cdouble (i.e. complex128) (CUDA is initialized)."
+                "Setting default tensor type to cuda.DoubleTensor (CUDA is initialized). "
+                "CUDA does not yet allow for complex tensors."
```

Comment:
I would remark in someway that CPU-based implementations are supported in torchquad, despite CUDA-based are currently not.